velox/functions/prestosql/tests/TimestampWithTimeZoneCastTest.cpp

```diff
@@ -64,6 +64,31 @@ TEST_F(TimestampWithTimeZoneCastTest, fromTimestamp) {
   testCast(tsVector, expected);
 }
 
+TEST_F(TimestampWithTimeZoneCastTest, fromVarchar) {
+  const auto stringVector = makeNullableFlatVector<StringView>(
+      {std::nullopt,
+       "2012-10-31 01:00:47 America/Denver",
+       "2012-10-31 01:00:47 -06:00",
+       "1979-02-24 08:33:31 Pacific/Chatham",
+       "1979-02-24 08:33:31 +13:45"});
+
+  auto timestamps = std::vector<int64_t>{
+      0,
+      1351666847 * kMillisInSecond,
```

Comment:
How did you get these numbers ? Is there a formula in Velox corresponding to this ? If yes, then we can write a test that compares the results of the cast and the expression. That would make this test more robust.

Reply:
I manually converted human-readable English dates to epoch values - though it would be a good idea to have a formula in Velox this, considering how widely such values are used in tests.

Reply:
Never mind, of course we have the existing timestamp conversion functions - used those to eliminate the magic numbers and clean up the test. Please take a look @aditi-pandit